generator/src/main/java/com/scottlogic/deg/generator/walker/DecisionTreeWalker.java

```diff
@@ -2,9 +2,10 @@
 
 import com.scottlogic.deg.generator.decisiontree.DecisionTree;
 import com.scottlogic.deg.generator.fieldspecs.RowSpec;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
 
 import java.util.stream.Stream;
 
 public interface DecisionTreeWalker {
-    Stream<RowSpec> walk(DecisionTree tree);
+    Stream<RowSpec> walk(DecisionTree tree, GenerationConfig config);
```

Comment:
I know we are getting rid of the Cartesian Walker but since the Reductive Walker is the only type that cares about this doesn't this highlight that perhaps GenerationConfig is not a general dependency?

Reply:
Agreed, it muddies the interface a bit. Reason for the change is 2 fold. A) we plan on removing the other walkers in the near future, as such there is little benefit in creating a new interface, where it pollutes the other implementations slightly it should be short lived. B) it is better that this non-dependency is passed in than held as a dependency when it's not - at least in the classical sense.

Reply:
parameter has been removed as part of a refactor